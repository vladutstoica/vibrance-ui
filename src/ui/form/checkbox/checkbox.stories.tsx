import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import { Label } from "../label/label";
import { Checkbox } from "./checkbox";

const meta: Meta<typeof Checkbox> = {
  title: "Design System/Form/Checkbox",
  component: Checkbox,
  tags: ["autodocs"],
  argTypes: {
    checked: {
      control: "boolean",
      description: "The controlled checked state of the checkbox",
      table: {
        type: { summary: "boolean" },
      },
    },
    defaultChecked: {
      control: "boolean",
      description: "The default checked state when initially rendered",
      table: {
        type: { summary: "boolean" },
      },
    },
    disabled: {
      control: "boolean",
      description:
        "When true, prevents the user from interacting with the checkbox",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    required: {
      control: "boolean",
      description:
        "When true, indicates that the user must check this checkbox",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Checkbox>;

// Common label styles
const labelStyles =
  "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70";

export const Default: Story = {
  render: (args) => (
    <div className="flex items-center space-x-2">
      <Checkbox {...args} />
      <Label htmlFor="terms">Accept terms and conditions</Label>
    </div>
  ),
  args: {
    id: "terms",
  },
};

export const WithText: Story = {
  render: (args) => (
    <div className="items-top flex space-x-2">
      <Checkbox {...args} />
      <div className="grid gap-1.5 leading-none">
        <Label htmlFor="terms1" className={labelStyles}>
          Accept terms and conditions
        </Label>
        <p className="text-muted-foreground text-sm">
          You agree to our Terms of Service and Privacy Policy.
        </p>
      </div>
    </div>
  ),
  args: {
    id: "terms1",
  },
};

export const Disabled: Story = {
  render: (args) => (
    <div className="flex items-center space-x-2">
      <Checkbox {...args} />
      <Label htmlFor="terms-disabled" className={labelStyles}>
        Accept terms and conditions
      </Label>
    </div>
  ),
  args: {
    disabled: true,
    id: "terms-disabled",
  },
};

export const DisabledChecked: Story = {
  render: (args) => (
    <div className="flex items-center space-x-2">
      <Checkbox {...args} />
      <Label htmlFor="terms-disabled-checked" className={labelStyles}>
        Accept terms and conditions
      </Label>
    </div>
  ),
  args: {
    disabled: true,
    defaultChecked: true,
    id: "terms-disabled-checked",
  },
};

export const Required: Story = {
  render: (args) => (
    <div className="flex items-center space-x-2">
      <Checkbox {...args} />
      <Label htmlFor="terms-required" className={labelStyles}>
        Accept terms and conditions{" "}
        <span className="text-red-500" aria-hidden="true">
          *
        </span>
      </Label>
    </div>
  ),
  args: {
    required: true,
    id: "terms-required",
    "aria-required": true,
  },
};
