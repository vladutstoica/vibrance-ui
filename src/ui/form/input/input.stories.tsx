import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Label } from "../label/label";
import { Input } from "./input";

const meta: Meta<typeof Input> = {
  title: "Design System/Form/Input",
  component: Input,
  tags: ["autodocs"],
  argTypes: {
    type: {
      description: "The type of the input field",
      control: "select",
      options: [
        "text",
        "email",
        "password",
        "number",
        "search",
        "tel",
        "url",
        "file",
      ],
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "text" },
      },
    },
    disabled: {
      description: "Whether the input is disabled",
      control: "boolean",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    placeholder: {
      description: "Placeholder text for the input",
      control: "text",
    },
  },
  decorators: [
    (Story, context) => (
      <div className="grid w-full max-w-sm gap-1.5">
        <Label htmlFor={`input-${context.name}`}>{context.name}</Label>
        {Story({ args: { ...context.args, id: `input-${context.name}` } })}
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    placeholder: "Enter your text",
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    placeholder: "Disabled input",
  },
};

export const WithValue: Story = {
  args: {
    defaultValue: "Hello World",
    "aria-label": "Text input with value",
  },
};

export const Email: Story = {
  args: {
    type: "email",
    placeholder: "Enter your email",
  },
};

export const Password: Story = {
  args: {
    type: "password",
    placeholder: "Enter your password",
  },
};

export const Search: Story = {
  args: {
    type: "search",
    placeholder: "Search...",
  },
};

export const File: Story = {
  args: {
    type: "file",
    accept: "image/*",
  },
};

export const WithError: Story = {
  render: (args) => (
    <div className="grid w-full max-w-sm gap-1.5">
      <Label htmlFor="input-error">Email</Label>
      <Input
        id="input-error"
        type="email"
        placeholder="Enter your email"
        className="border-red-500 focus-visible:ring-red-500"
        aria-describedby="email-error"
        {...args}
      />
      <p id="email-error" className="text-red-500 text-sm">
        Please enter a valid email address
      </p>
    </div>
  ),
};
