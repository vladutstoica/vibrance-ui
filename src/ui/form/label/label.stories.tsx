import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Checkbox } from "../checkbox/checkbox";
import { Input } from "../input/input";
import { Label } from "./label";

const meta: Meta<typeof Label> = {
  title: "Form/Label",
  component: Label,
  tags: ["autodocs"],
  argTypes: {
    htmlFor: {
      description: "The id of the form control this label is associated with",
      control: "text",
    },
  },
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Default: Story = {
  args: {
    children: "Accept terms and conditions",
  },
};

export const WithInput: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="email">Your email address</Label>
      <Input type="email" id="email" placeholder="Email" />
    </div>
  ),
};

export const WithCheckbox: Story = {
  render: () => (
    <div className="flex items-center space-x-2">
      <Checkbox id="terms" />
      <Label htmlFor="terms">Accept terms and conditions</Label>
    </div>
  ),
};

export const Required: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="required-email">
        Email <span className="text-red-500">*</span>
      </Label>
      <Input type="email" id="required-email" placeholder="Email" required />
    </div>
  ),
};

export const WithDescription: Story = {
  render: () => (
    <div className="grid w-full max-w-sm gap-1.5">
      <Label htmlFor="picture">Picture</Label>
      <Input id="picture" type="file" />
      <p className="text-muted-foreground text-sm">
        Upload a picture to your profile.
      </p>
    </div>
  ),
};
