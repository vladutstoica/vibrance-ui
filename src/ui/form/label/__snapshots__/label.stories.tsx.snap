// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form/Label Default smoke-test 1`] = `
<label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70">
  Accept terms and conditions
</label>
`;

exports[`Form/Label Required smoke-test 1`] = `
<div class="grid w-full max-w-sm items-center gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="required-email"
  >
    Email
    <span class="text-red-500">
      *
    </span>
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         id="required-email"
         placeholder="Email"
         required
         type="email"
  >
</div>
`;

exports[`Form/Label WithCheckbox smoke-test 1`] = `
<div class="flex items-center space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="false"
          data-state="unchecked"
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms"
  >
  </button>
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="terms"
  >
    Accept terms and conditions
  </label>
</div>
`;

exports[`Form/Label WithDescription smoke-test 1`] = `
<div class="grid w-full max-w-sm gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="picture"
  >
    Picture
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         id="picture"
         type="file"
  >
  <p class="text-muted-foreground text-sm">
    Upload a picture to your profile.
  </p>
</div>
`;

exports[`Form/Label WithInput smoke-test 1`] = `
<div class="grid w-full max-w-sm items-center gap-1.5">
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="email"
  >
    Your email address
  </label>
  <input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:font-medium file:text-foreground file:text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
         id="email"
         placeholder="Email"
         type="email"
  >
</div>
`;
